### ISSUE TITLE
JavadocStyle: check if javadoc starts with an uppercase letter

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#3468]
I would like a new feature in `JavadocStyle` where it checks that if the first word of the JavaDoc starts with a letter, it is capitalized.
This check already verifies if the JavaDoc contains a period of some kind. Periods ensures that it is a sentence, and in most languages, proper sentences usually start with an uppercase letter.

````
$ cat TestClass.java
public class TestClass {
    /** this is a method. */
    void method1() {}
    /** 101 leagues under the sea. */
    void method2() {}
    /** <b>this is a method.</b> */
    void method3() {}
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="JavadocStyle" />
    </module>
</module>

$ java -jar checkstyle-7.1.1-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````

I expect violations on `method1` and `method3` because the first word of the sentence, `this`, is not capitalized. There should be no violations if the user wrote `This` in the JavaDoc.
`method2` has no violation because it's first word `101` doesn't not start with a letter.

**Comment by kukreja-vikramaditya** [comment]
I am on it

**Comment by rnveach** [comment]
@kukreja-vikramaditya This issue doesn't have approve label on it, so it can't be worked on until then.

**Comment by kukreja-vikramaditya** [comment]
Should this new property be able to check for character inside a tag? <- thought for future work on this

-----

Javadoc 1
 ```java
/** <b>hello</b> */
```
Javadoc 2
```java
/**
 * <pre>
 * class Foo {
 *
 *   void foo() {}
 * }
 * </pre>
 */
```
Is it possible to differentiate between the 2 with some sort of javadoc token detector? What if user has custom tags?

**Comment by rnveach** [comment]
> Should this new property be able to check for character inside a tag?

IMO, as long as it is part of the first sentence of the javadoc description, it should validate that text inside.
We should use `javadoc` utility to confirm all tags and first sentences.

These are my thoughts:
`<b>` - Bolding/underlining/etc is just to make text flashy, First sentence is inside tag and should still be validated.
`<pre>` - To me, `pre` shouldn't be considered part of first sentence.
`<p>` - Same as bolding, first text is inside.
`<table>` - Same as `pre`.
`<ul>` - Same as `pre`.

> Is it possible to differentiate between the 2 with some sort of javadoc token detector?

Use the Javadoc tree, yes. I doubt the regular expression version of the check could.

> What if user has custom tags?

Must see how `javadoc` utility handles it, then we could decide.

**Comment by jdussouillez** [comment]
Any news about this feature ? Or a workaround ?

**Comment by ralienpp** [comment]
Considering that the last activity here was 3 years ago, I'd like "rekindle the flame". Having such a check would make it easier to publish well-polished documentation.

**Comment by romani** [comment]
JavadocStyle is removed from Checkstyle at 13.9.0, users should use SummaryJavadoc instead
Details:  https://github.com/checkstyle/checkstyle/issues/20610
